packages/mdc-tab-bar/foundation.js

```diff
@@ -0,0 +1,365 @@
+    const leftIncrement = relativeContentRight + numbers.EXTRA_SCROLL_AMOUNT;
+    const rightIncrement = relativeContentLeft - numbers.EXTRA_SCROLL_AMOUNT;
+
+    if (nextIndex > index) {
+      return Math.max(leftIncrement, 0);
+    }
+
+    return Math.min(rightIncrement, 0);
+  }
+
+  /**
+   * Determines the index of the next adjacent tab
+   * @param {number} index The index of the tab
+   * @param {!MDCTabDimensions} tabDimensions Whether the next tab is left or right
+   * @param {number} scrollPosition The current scroll position
+   * @param {number} barWidth The width of the tab bar
+   * @return {number}
+   * @private
+   */
+  findTabIndexClosestToEdge_(index, tabDimensions, scrollPosition, barWidth) {
+    /**
+     * Tabs are laid out in the Tab Scroller like this:
+     *
+     *    Scroll Position
+     *    +---+
+     *    |   |   Bar Width
+     *    |   +-----------------------------------+
+     *    |   |                                   |
+     *    |   V                                   V
+     *    |   +-----------------------------------+
+     *    V   |             Tab Scroller          |
+     *    +------------+--------------+-------------------+
+     *    |    Tab     |      Tab     |        Tab        |
+     *    +------------+--------------+-------------------+
+     *        |                                   |
+     *        +-----------------------------------+
+     *
+     * To determine the next adjacent index, we look at the Tab root left and
+     * Tab root right, both relative to the scroll position. If the Tab root
+     * left is less than 0, then we know it's out of view to the left. If the
+     * Tab root right minues the bar width is greater than 0, we know the Tab is
+     * out of view to the right. From there, we either increment or decrement
+     * the index.
+     */
+    const relativeRootLeft = tabDimensions.rootLeft - scrollPosition;
+    const relativeRootRight = tabDimensions.rootRight - scrollPosition - barWidth;
+    const relativeRootDelta = relativeRootLeft + relativeRootRight;
+    const leftEdgeIsCloser = relativeRootLeft < 0 || relativeRootDelta < 0;
+    const rightEdgeIsCloser = relativeRootRight > 0 || relativeRootDelta > 0;
+
+    if (leftEdgeIsCloser) {
+      return index - 1;
+    }
+
+    if (rightEdgeIsCloser) {
+      return index + 1;
+    }
+
+    return -1;
+  }
+
+  /**
+   * Determines the index of the next adjacent tab in RTL
+   * @param {number} index The index of the tab
+   * @param {!MDCTabDimensions} tabDimensions Whether the next tab is left or right
+   * @param {number} scrollDistance The distance scrolled
+   * @param {number} scrollWidth The width of the scroller
+   * @param {number} barWidth The width of the bar
+   * @return {number}
+   * @private
+   */
+  findTabIndexClosestToEdgeRTL_(index, tabDimensions, scrollDistance, scrollWidth, barWidth) {
```

Comment:
I'm not sure why some parameter and variable names here are different from the non-RTL counterpart; it seems like the same things are passed in either case. Can we sync these up?